hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -132,6 +132,7 @@
 	 */
 	private final boolean isKeyTableNullable;
 	private final boolean isConstrained;
+	private final boolean isForeignKeyEnabed;
```

Comment:
Actually, `isConstrained` should be enough. I don't think you need another field, or am I missing something?

Reply:
unfortunately for a `OneToOne` is not, I tried to use it but a I had a lot of failures :(

Reply:
@beikov by the way it seems this solution is not working on top of the latest changes... I'm working on a slightly different solution